scripts/search.js

```diff
@@ -0,0 +1,153 @@
+      // intercept the query and slice of api: before searching
+      types.forEach(type => {
+        if (query.includes(`${type.name}:`)) {
+          let filteredTypes = types.filter(ele => ele.name !== type.name)
+          console.log(filteredTypes)
+          filteredTypes.forEach(filteredType => {
+            document.getElementById(`${filteredType.name}-hits`).style.display = 'none'
+          })
+          document.getElementById(`${type.name}-hits`).style.height = 'auto'
+          document.getElementById(`${type.name}-hits`).style.overflow = 'scroll'
+          query = query.slice(query.indexOf(':'))
+          typesToSearch = [{name: `${type.name}`}]
+        }
+      })
+
+      typesToSearch.forEach(type => {
+        const search = searches[type.name]
+        if (search && search.helper) {
+          search.helper.setQuery(query)
+          search.helper.search()
+        }
+      })
+
+      helper.search()
+    }
+  }
+
+  const search = instantsearch(opts)
+
+  search.addWidget(
+    instantsearch.widgets.hits({
+      container: `#${type}-hits`,
+      templates: {
+        empty: 'No results',
+        item: templates[type]
+      },
+      transformData: {
+        item: data => {
+          // useful for viewing template context:
+          // console.log(`${type} data`, data)
+          return data
+        }
+      }
+    })
+  )
+
+  if (isPrimarySearch) {
+    search.addWidget(
+      instantsearch.widgets.searchBox({
+        container: '#search-input',
+        placeholder: `Search Electron ${pluralize(type)}`
+      })
+    )
+  }
+
+  search.on('render', (...args) => {
+    // console.log(`algolia render (${type})`, args)
+  })
+
+  search.on('error', (...args) => {
+    console.log(`algolia error (${type})`, args)
+  })
+
+  searches[type] = search
+
+  search.start()
+}
+
+function determineOrder () {
+  types.forEach(type => {
+    if (location.pathname === type.path) document.getElementById(`${type.name}-hits`).className = 'first'
+  })
+}
+
+// function buildSoloSearch(){
+//   //loop over types obj in search of current pathname
+//   if(location.pathname === '/docs/api'){
+//     console.log('yay!')
+//     buildSearch('api', true, true)
+//   }
+// }
+
+function buildMultiSearch () {
+  types.forEach(type => {
+    buildSearch(type.name, type.name === types[0].name, searches)
+  })
+}
+
+function buildSearchUIHandlers () {
+  let navInput = document.querySelector('.nav-search')
+  let hits = document.getElementById('hits')
+  navInput.addEventListener('input', e => showHits())
+  navInput.addEventListener('focus', e => showHits())
+  window.addEventListener('click', e => {
+    e.target === navInput || checkIfChild(hits, e.target) ? showHits() : hideHits()
+  })
+
+  function showHits () {
+    document.getElementById('hits').style.display = 'flex'
+  }
+
+  function hideHits () {
+    document.getElementById('hits').style.display = 'none'
+  }
+
+  function checkIfChild (parentElement, checkElement) {
```

Comment:
@zeke HALP. trying to recursively check for any child of "hits" to fix click functionality. something is broken. Line 135 calls this function.